mgmt/rest/v2/task.go

```diff
@@ -33,26 +37,89 @@ import (
 	"github.com/julienschmidt/httprouter"
 )
 
+// TasksResp returns a list of created tasks.
+//
+// swagger:response TasksResponse
+type TasksResp struct {
+	// in: body
+	Body struct {
+		Tasks Tasks `json:"tasks"`
+	}
+}
+
+// TaskResp returns the giving task.
+//
+// swagger:response TaskResponse
+type TaskResp struct {
+	// in: body
+	Task Task `json:"task"`
+}
+
+// RemoveTaskError unsuccessful generic response to a failed API call
+//
+// swagger:response TaskErrorResponse
+type RemoveTaskError struct {
+	// in: body
+	Message string `json:"message"`
+}
+
+// TasksResponse returns a list of created tasks.
 type TasksResponse struct {
 	Tasks Tasks `json:"tasks"`
 }
 
+// TaskParam contains task id.
+//
+// swagger:parameters getTask watchTask updateTaskState removeTask
+type TaskParam struct {
+	// in: path
+	// required: true
+	ID string `json:"id"`
+}
+
+// TaskPostParams defines task POST and PUT entity.
+// swagger:parameters addTask
+type TaskPostParams struct {
+	// Create or update a task.
+	//
+	// in: formData
+	// required: true
+	Task string `json:"task"yaml:"task"`
+}
+
+// TaskPutParams defines the type for updating a task.
+//
+// swagger:parameters updateTaskState
+type TaskPutParams struct {
+	// Update the state of a task
+	//
+	// in: formData
+	//
+	// required: true
+	Action string `json:"action"`
+}
+
+// Task represents Snap task definition.
 type Task struct {
-	ID                 string            `json:"id"`
-	Name               string            `json:"name"`
-	Deadline           string            `json:"deadline"`
-	Workflow           *wmap.WorkflowMap `json:"workflow,omitempty"`
-	Schedule           *core.Schedule    `json:"schedule,omitempty"`
-	CreationTimestamp  int64             `json:"creation_timestamp,omitempty"`
-	LastRunTimestamp   int64             `json:"last_run_timestamp,omitempty"`
-	HitCount           int               `json:"hit_count,omitempty"`
-	MissCount          int               `json:"miss_count,omitempty"`
-	FailedCount        int               `json:"failed_count,omitempty"`
-	LastFailureMessage string            `json:"last_failure_message,omitempty"`
-	State              string            `json:"task_state"`
-	Href               string            `json:"href"`
+	// unique: true
```

Comment:
A very good question. Actually, we should annotate all required and unique fields. Only when we use a string input, it puts those annotations as unnecessary but they're beneficial for generated clients in different languages. Therefore adding more such tedious annotations in the future may be necessary.